arroyo-worker/src/formats/mod.rs

```diff
@@ -180,22 +180,22 @@ impl<T: SchemaData> DataSerializer<T> {
     pub fn to_vec(&self, record: &T) -> Option<Vec<u8>> {
         match &self.format {
             Format::Json(json) => {
-                let v = if json.include_schema {
+                let mut writer: Vec<u8> = vec![];
```

Comment:
For performance, I think we should follow the serde_json::to_vec behavior of preallocating 128 bytes here; otherwise you could end up repeatedly reallocating the buffer as you write

Reply:
As the initial size? sure.